optimization.jl

```diff
@@ -0,0 +1,115 @@
+using Turing
+using StatsPlots
+# solar mass and gravitational constant in CGS
+const cgrav = 6.67430e-8
+const mu_sun = 1.3271244e26
+const m_sun = mu_sun/cgrav
+const day_in_sec = 24*3600
+
+##
+"""
+    post_kick_parameters_P(P,m1,m2i,m2f,vkick)
+
+Compute post-kick properties using equations from Tauris et al. (1999), but using the orbital period
+
+#Arguments:
+- P:     pre-explosion orbital period               [d]
+- mtilde: (m1_i+m2_f)/(m1_i+m2_i) [unitless], here we have
+    - m1:    mass of non-exploding component
+    - m2i:   mass of exploding component
+    - m2f:   post explosion mass of exploding component
+- vkick_div_vrel: kick velocity in units of the pre-explosion relative velocity    [unitless]
+    - vrel = sqrt(G(m1_i + m2_i)/a)
+- theta: polar angle indicating kick direction      [rad]
+- phi:   azimuthal angle indicating kick direction  [rad]
+
+#Output: (Ppost, epost) where
+- Ppost: post-explosion orbital period              [d]
+- epost: post-explosion eccentricity                [dimensionless]
+"""
+function post_kick_parameters_P(P,mtilde,vkick_div_vrel,theta,phi)
+    xi = (1+vkick_div_vrel^2+2*vkick_div_vrel*cos(theta))/(mtilde)
+    Q = xi - 1 - (vkick_div_vrel*sin(theta)*cos(phi))^2/(mtilde)
+
+    if (xi>2)
+        return (NaN*P, NaN*P) # multiply by P to make outcome type stable
+    end
+    Pfinal = P/sqrt(mtilde*(2-xi)^3)
+    efinal = sqrt(1+(xi-2)*(Q+1))
+    return (Pfinal, efinal)
+end
+
+##
+
+typeof(post_kick_parameters_P)
+first(methodswith(Function), 5)
```

Comment:
These two should go

Reply:
Wait, I just realized that this is in the optimization.jl file. We probably dont want to merge this  file to main